core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -882,8 +882,9 @@ private boolean analyzeListOfValues(
                 return false;
             }
 
-            //if key values contain bind variable then we can't merge it with any other set and have to push this list to filter
-            if (!allKeyValuesAreKnown) {
+            // if key values contain bind variable then we can't merge it with any other set and have to push this list to filter,
+            // we can only create a new list of keys (=newColumn)
+            if (!allKeyValuesAreKnown & !newColumn) {
                 return false;
             }
 
```

Comment:
nit: the code in the below `if (i == 1)` branch breaks on `? = col` instead of `col = ?`.